### ISSUE TITLE
Parsing error when receiving an unknown type for a union

### ISSUE THREAD

**Issue by Mirconet** [apollographql/apollo-ios#2011]
## Bug report

When the schema defines a union with a single type, the generated swift code fails to parse a new unknown type of the union from server. This doesn't happen if we map the union with a fragment: the generated swift code now manage the new unknown type correctly.

## Versions
- `apollo-ios` SDK version: 0.49.1
- Xcode version: 12.5.1

## Steps to reproduce
With the following schema:

```
union Discount = DiscountMotor

type DiscountMotor {
  description: String!
}
```

if I define the query in this way:
```
appliedDiscounts {
    ... on DiscountMotor {
        description
    }
}
```
the generated code is
```
public struct AppliedDiscount: GraphQLSelectionSet {
      public static let possibleTypes: [String] = ["DiscountMotor"]

      public static var selections: [GraphQLSelection] {
        return [
          GraphQLField("__typename", type: .nonNull(.scalar(String.self))),
          GraphQLField("description", type: .nonNull(.scalar(String.self))),
        ]
      }
```
When the server adds a new type to the union:
```
union Discount = DiscountMotor | DiscountFamily

type DiscountFamily {
  description: String!
}
```
the app with the old schema return this error:
`Apollo.GraphQLResultError(path: appliedDiscounts.0.description, underlying: Apollo.JSONDecodingError.missingValue)`

This doesn't happen if I use a fragment to map the union like this:
```
appliedDiscounts {
    ... DiscountFragment
}

fragment DiscountFragment on Discount {
    ... on DiscountMotor {
        description
    }
}
```
This was the generated code in this case:
```
public struct AppliedDiscount: GraphQLSelectionSet {
      public static let possibleTypes: [String] = ["DiscountMotor"]

      public static var selections: [GraphQLSelection] {
        return [
          GraphQLField("__typename", type: .nonNull(.scalar(String.self))),
          GraphQLFragmentSpread(DiscountFragment.self),
        ]
      }
```
and the app with the old schema can ignore the unknown type correctly.

**Comment by andreasreuter** [comment]
I get the same problem. codgen always adds ".notNull" and it doesn't care about nullable fields. I use this script
```
SCRIPT_PATH="${PODS_ROOT}/Apollo/scripts"
cd "${SRCROOT}/${TARGET_NAME}"
"${SCRIPT_PATH}"/run-bundled-codegen.sh codegen:generate --target=swift --includes=./**/*.graphql --localSchemaFile="schema.json" Networking/API.swift
```

**Comment by AnthonyMDev** [comment]
Luckily, this won't be a problem with the new generated code. We're getting really close to a 1.0 release!

**Comment by calvincestari** [comment]
Hi 👋🏻 - this issue should be resolved in 1.0. 1.0 is currently in alpha releases with a full 1.0 release in H2 2022. Please take a look at the [latest alpha](https://github.com/apollographql/apollo-ios/releases) and let us know if this is still an issue for you.
